tests/test_validate.py

```diff
@@ -525,5 +146,350 @@ def test_source_violaiton():
+
+def test_find_nested_ib_violations_mbz(
+    tmp_path, mbz_builder, page_builder, mocker
+):
+    bad_content = '<div><div class="os-raise-ib-nestedtype"></div></div>'
+    page1 = page_builder(1, "Page", bad_content)
+    mbz_builder(tmp_path / "mbz", [page1])
+
+    mocker.patch(
+        "sys.argv",
+        ["", f"{tmp_path}/mbz", f"{tmp_path}/test_output.csv"]
+    )
+
+    validate_mbz_html.main()
+
+    reader = csv.DictReader(open(tmp_path / "test_output.csv"))
+    errors = [row for row in reader]
+    assert len(errors) == 1
+    assert errors[0]["issue"] == validate_mbz_html.NESTED_IB_VIOLATION
+    assert errors[0]["location"] == "Page"
+    assert errors[0]["link"] == "os-raise-ib-nestedtype"
+
+
+def test_find_tag_violations_mbz(
+    tmp_path, mbz_builder, page_builder, mocker
+):
+    bad_content = '<div><script></script></div>'
+    page1 = page_builder(1, "Page", bad_content)
+    mbz_builder(tmp_path / "mbz", [page1])
+
+    mocker.patch(
+        "sys.argv",
+        ["", f"{tmp_path}/mbz", f"{tmp_path}/test_output.csv"]
+    )
+
+    validate_mbz_html.main()
+
+    reader = csv.DictReader(open(tmp_path / "test_output.csv"))
+    errors = [row for row in reader]
+    assert len(errors) == 1
+    assert errors[0]["issue"] == validate_mbz_html.SCRIPT_VIOLATION
+    assert errors[0]["location"] == "Page"
+    assert errors[0]["link"] == ""
+
+
+def test_find_style_violations_mbz(
+    tmp_path, mbz_builder, page_builder, mocker
+):
+    bad_content = '<p style="color: blue"></p>'
+    page1 = page_builder(1, "Page", bad_content)
+    mbz_builder(tmp_path / "mbz", [page1])
+
+    mocker.patch(
+        "sys.argv",
+        ["", f"{tmp_path}/mbz", f"{tmp_path}/test_output.csv"]
+    )
+
+    validate_mbz_html.main()
+
+    reader = csv.DictReader(open(tmp_path / "test_output.csv"))
+    errors = [row for row in reader]
+    assert len(errors) == 1
+    assert errors[0]["issue"] == validate_mbz_html.STYLE_VIOLATION
+    assert errors[0]["location"] == "Page"
+    assert errors[0]["link"] == "color: blue"
+
+
+def test_find_source_violations_mbz(
+    tmp_path, mbz_builder, page_builder, mocker
+):
+    bad_content = '<img src="http://foobar">'
+    page1 = page_builder(1, "Page", bad_content)
+    mbz_builder(tmp_path / "mbz", [page1])
+
+    mocker.patch(
+        "sys.argv",
+        ["", f"{tmp_path}/mbz", f"{tmp_path}/test_output.csv"]
+    )
+
+    validate_mbz_html.main()
+
+    reader = csv.DictReader(open(tmp_path / "test_output.csv"))
+    errors = [row for row in reader]
+    assert len(errors) == 1
+    assert errors[0]["issue"] == validate_mbz_html.SOURCE_VIOLATION
+    assert errors[0]["location"] == "Page"
+    assert errors[0]["link"] == "http://foobar"
+
+
+def test_find_multiple_activity_violations_mbz(
+    tmp_path, mbz_builder, page_builder, lesson_builder, mocker
+):
+    bad_content = '<img src="http://foobar"><script></script>'
+    page1 = page_builder(1, "Page", bad_content)
+    lesson2 = lesson_builder(
```

Comment:
Oh, I think this is just some confusion around what `pytest` is giving when you use a fixture: It provides the return of the function used to implement the fixture. So that's why our fixture returns a function which takes parameters we want to use to generate something. When we do `test_foobar(lesson_builder)` pytest is going to return the function returned by `lesson_builder` which is then in scope for the test function. What we couldn't do, for example, is have the parameters go directly to `test_builder` as pytest expects those to be references to other fixtures, not parameters to the fixture itself. You can see an example of that syntax as well in `lesson_builder` where it takes as input the `lesson_page_builder` fixture to use within its own definition.